### ISSUE TITLE
Can't build containers on macOS after upgrade to 0.23.0

### ISSUE THREAD

**Issue by eikemeier** [bazelbuild/rules_docker#2009]
# 🐞 bug report

### Affected Rule

The issue is caused by the rule: `java_image`

### Is this a regression?

Yes, the previous version in which this bug was not present was: 0.22.0

### Description

When building a simple Java Docker image on macOS, the build fails with “No matching toolchains found”. It works on Linux, since the target and host platforms are identical.

## 🔬 Minimal Reproduction

See https://github.com/eikemeier/java-container

## 🔥 Exception or Error

<pre><code>
bazel build //src/main/java/com/example/hello_world:hello_world_image
[...]
ERROR: [...]/java-container/src/main/java/com/example/hello_world/BUILD:16:11: While resolving toolchains for target //src/main/java/com/example/hello_world:hello_world_image.binary: No matching toolchains found for types @bazel_tools//tools/cpp:toolchain_type. [...]
</code></pre>

## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
macOS Catalina 10.15.7
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
5.0.0
  </code>
</pre>

**Rules_docker version:**

<pre>
  <code>
0.23.0
  </code>
</pre>

**Anything else relevant?**

See also https://github.com/bazelbuild/rules_docker/issues/1988#issuecomment-1016465174

**Comment by dragonsinth** [comment]
Same for `go_image`; 0.22.0 works, 0.23.0 doesn't.  According to `git bisect`, seems to be introduced in:
Transition container image target platform (#1963) 76c708fc979c1bfb65b4db300c654be08f096874

**Comment by pcmoritz** [comment]
Same for `py3_image`, 0.22.0 works and 0.23.0 doesn't. Hopefully this can be fixed soon :)

**Comment by psalaberria002** [comment]
Is anyone looking into this issue? Any workaround other than going back to 0.22?

**Comment by shs96c** [comment]
We've had to prepare a cross-compiling C toolchain to use on macOS in order to make this work. Ultimately, we picked one based on musl, since that's most likely to work on most linux variants, but we've also had to patch `rules_go` to force `go_binary` to statically link when building for a docker image. Getting this working properly has been a pretty significant piece of work.

If anyone is following in our footsteps, we started off from [musl-cross-make](https://github.com/richfelker/musl-cross-make)

**Comment by kylekurz** [comment]
I don't think this is closable, I'm still locked to 0.22 waiting on a fix for this.

**Comment by trungdinhth** [comment]
With `py3_image`, for 0.25 we are still having this problem.
